### PR TITLE
[Keyboard]Add X-Bows Knight_Plus Keyboard

### PR DESCRIPTION
**Pull request by XBowsTech**
## Description

Add support for X-Bows Knight_Plus Keyboard and Numpad Keyboard.
This is Knight_Plus keyboard set that contains Knight_Plus and Numpad.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/xbows/knight_plus/knight_plus.c [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+
+void suspend_power_down_kb(void) {
+    rgb_matrix_set_suspend_state(true);
+    suspend_power_down_user();
+}
+
+void suspend_wakeup_init_kb(void) {
+    rgb_matrix_set_suspend_state(false);
+    suspend_wakeup_init_user();
+}
```
Comment (drashna) [inline keyboards/xbows/knight_plus/knight_plus.c:146]:
```suggestion
```

File: keyboards/xbows/numpad/numpad.c [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+void suspend_power_down_kb(void) {
+    rgb_matrix_set_suspend_state(true);
+    suspend_power_down_user();
+}
+
+void suspend_wakeup_init_kb(void) {
+    rgb_matrix_set_suspend_state(false);
+    suspend_wakeup_init_user();
+}
+
```
Comment (drashna) [inline keyboards/xbows/numpad/numpad.c:83]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/xbows/knight_plus/knight_plus.c (hunk 1 above)
Comment (XBowsTech) [reply keyboards/xbows/knight_plus/knight_plus.c:146]: Why delete them?

Comment (drashna) [reply keyboards/xbows/knight_plus/knight_plus.c:146]: This is handled in the core code, now.  You only need the define to enable the functionality, much like `RGBLIGHT_SLEEP`.

https://github.com/qmk/qmk_firmware/blob/89c01970e333e346c25202e9c568b75cad4c7be5/tmk_core/common/avr/suspend.c#L162-L171

**Review by drashna — APPROVED** [review]

**Comment by XBowsTech** [comment]
I added the requested changes. @fauxpark

**Comment by XBowsTech** [comment]
I added the requested changes @zvecr

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**